distributed/tensor_parallelism/two_d_parallel_example.py

```diff
@@ -54,74 +53,109 @@
 """
 
 
-def demo_2d(rank, args):
+def demo_2d(args):
     """
     Main body of the demo of a basic version of tensor parallel by using
     PyTorch native APIs.
     """
-    print(f"Running basic Megatron style TP example on rank {rank}.")
-    setup(rank, args.world_size)
+    torchrun_setup()
+
+    # understand world topology
+    _rank = int(os.environ["RANK"])
+    _local_rank = int(os.environ["LOCAL_RANK"])
+    _world_size = int(os.environ["WORLD_SIZE"])
+    _local_world_size = int(os.environ["LOCAL_WORLD_SIZE"])
+
+    torch.cuda.set_device(_local_rank)
```

Comment:
similar comments as the `sequence_parallel_example.py`, please address :)

Reply:
all better.